File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.py
class Solution:
    def computeGCD(self,x, y):
        while(y):
            x, y = y, x % y
        return abs(x)
    def minOperations(self, nums: List[int], numsDivide: List[int]) -> int:
        ans=numsDivide[0]
        for i in numsDivide:
            ans=self.computeGCD(ans,i)
        nums.sort()
        # print(ans,nums)
        cnt=0
        for i in nums:
            if ans%i==0:
                return cnt
            cnt+=1
        return -1
        

File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, vector<int>& numsDivide) {
        int lc=numsDivide[0];
        for(int i=0;i<numsDivide.size();i++){
            lc=__gcd(lc,numsDivide[i]);
        }
        // cout<<lc<<endl;
        int cnt=0;
        sort(nums.begin(),nums.end());
        for(int &i:nums){
            if(i>lc)
                break;
            if(lc%i==0){
                return cnt;
            }
            cnt++;
        }
        return -1;
    }
};